code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -62,9 +64,27 @@
 	succeed_activate()
 
 	X.throw_at(A, RAV_CHARGEDISTANCE, RAV_CHARGESPEED, X)
+	activate_particles(TRUE, X.dir)
 
 	add_cooldown()
 
+/datum/action/xeno_action/activable/charge/proc/activate_particles(boolean = FALSE, direction)
+	if(!boolean)
+		QDEL_NULL(particle_holder)
+		return
+	particle_holder = new(get_turf(owner), /particles/ravager_charge)
+	addtimer(CALLBACK(src, .proc/activate_particles), 5) // Boolean is false by default, so this deactivates particles.
+	particle_holder.particles.rotation += dir2angle(direction)
```

Comment:
Doesn't using dir mean the effect won't be facing the right direction if you use this in a non cardinal direction?

Reply:
Doesn't seem to be the case. I definitely tested non-cardinal directions, and everything seemed fine.

Reply:
This aged like milk. Which is to say, awfully.